contrib/exporters/core/extend.go

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package core
+
+import (
+	"strings"
+	"encoding/json"
+	"github.com/spf13/viper"
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/api/client"
+)
+
+// for no transform extensions
+type extendNone struct {
+}
+
+func NewExtendNone(cfg *viper.Viper) (interface{}, error) {
+	logging.GetLogger().Infof("entering NewExtendNone")
+        return &extendNone{}, nil
+}
+
+func (e *extendNone)  Extend(in interface{}) interface{} {
+	return in
+}
+
+// for transform extensions
+// Syntax supported is:
+//	VAR_NAME=<gremlin expression with a single substitution string>
+// For example:
+//	AA_Name=G.V().Has('RoutingTables.Src',%Network.A%).Values('Host')
+// Substitution string must be surrounded with `%`
+// The substitution string refers to fields that exist in the data provided by the particular transform
+// No spaces allowed
+// Need to define `extend` field as a subfield of `transform`
+// Structure in yml file is as follows:
+//   transform:
+//     type: <xxx>
+//     extend:
+//       - VAR_NAME1=<gremlin expression with a single substitution string>
```

Comment:
calculating a value via a single Gremlin expression will not work in the general case and would even not suffice the SecAdvisor and other cases:

```
// IPToName resolve ip to name
func (r *resolveRunc) IPToName(ipString, nodeTID string) (string, error) {
        node, err := r.gremlinClient.GetNode(g.G.V().Has("Runc.Hosts.IP", ipString).ShortestPathTo(g.Metadata("TID", nodeTID)))
        if err != nil {
                return "", err
        }

        name := node.Metadata["Runc"].(map[string]interface{})["Hosts"].(map[string]interface{})["Hostname"].(string)
        return "0_0_" + name + "_0", nil
}
```



this will not work in the general case - as you may need to run multiple Gremlin for example:

Reply:
Multiple Gremlin expressions can be defined, one after another, to define whatever is needed. Do you mean we need to combine multiple Gremlin expressions in a single query? This is a first implementation with initial functionality. We should discuss what the additional requirements are and see how to address them with improvements over time.

Reply:
To calculate this particular name (not including the prefix 0_0_, etc), you can do:
G.V().Has("Runc.Hosts.IP", ipString).ShortestPathTo(g.Metadata("TID", nodeTID)).Values('Runc.Hosts.Hostname)

Reply:
ok. For now lets' see if we can cover needed functionality w/o requiring multiple Gremlin queries.